Parsers/Epic_Parser.csl

```diff
@@ -0,0 +1,88 @@
+// Epic SIEM
+// Last Updated Date: June 2, 2020
+//
+// This parser parses Syslogs event from the Epic SIEM platform.
+// https://www.ibm.com/support/knowledgecenter/en/SS42VS_DSM/com.ibm.dsm.doc/c_dsm_guide_epic_securtiy_siem_overview.html?cp=SS42VS_7.3.3
+//
+// Parser Notes:
+// 1. This parser assumes logs are collected into the CommonSecurityLog table.
+//
+// Usage Instruction : 
+// Paste below query in log analytics, click on Save button and select as Function from drop down by specifying function name and alias.
+// Functions usually takes 10-15 minutes to activate. You can then use function alias from any other queries (e.g. Epic_CL | take 10).
+//
+//
+
+CommonSecurityLog
+| where DeviceVendor == "Epic"
+| extend AdditionalExtensions = replace(@'\^', @'/', AdditionalExtensions)
```

Comment:
Let`s add a check for all these column exists and create it with value None if not present since not all columns appear .

I created this piece of code that you can insert before line 18.

```
| extend SourceUserName=columnifexists("SourceUserName","None"),  workstationID=columnifexists("workstationID","None"),  end=columnifexists("end","None"),  flag=columnifexists("flag","None"), 
 AUDITSESSION=columnifexists("AUDITSESSION","None"),  BTGREASON=columnifexists("BTGREASON","None"),  CSN=columnifexists("CSN","None"),  DAT=columnifexists("DAT","None"),  DEP=columnifexists("DEP","None"),  PATIENT=columnifexists("PATIENT","None"), 
  BTGNOACCESSREAS=columnifexists("BTGNOACCESSREAS","None"),  AUTHSOURCE=columnifexists("AUTHSOURCE","None"),  LOGINCONTEXT=columnifexists("LOGINCONTEXT","None"),  LOGINDEVICE=columnifexists("LOGINDEVICE","None"),  LOGINLDAPID=columnifexists("LOGINLDAPID", "None"), 
   LOGINREVAL=columnifexists("LOGINREVAL", "None"),  NEWDEPARTMENT=columnifexists("NEWDEPARTMENT", "None"),  PREVDEPARTMENT=columnifexists("PREVDEPARTMENT", "None"),  HKUAPVER=columnifexists("HKUAPVER", "None"),  HKUDVCID=columnifexists("HKUDVCID", "None"),  
   HKUOSNAM=columnifexists("HKUOSNAM", "None"),  HKUOSVER=columnifexists("HKUOSVER", "None"),  LOGINERROR=columnifexists("LOGINERROR", "None"),  ERRMSG=columnifexists("ERRMSG", "None"),  PWREASON=columnifexists("PWREASON", "None"),  SUCCESS=columnifexists("SUCCESS", "None"), 
    UID=columnifexists("UID", "None"),  src_dest_IPs=columnifexists("src_dest_IPs", "None"),  PRTCTDSRCUSERID=columnifexists("PRTCTDSRCUSERID", "None"),  WEBLGAPP=columnifexists("WEBLGAPP", "None"),  SOURCE=columnifexists("SOURCE", "None"),  APIID=columnifexists("APIID", "None"),  
    APPLICATIONID=columnifexists("APPLICATIONID", "None"),  INSTANCEURN=columnifexists("INSTANCEURN", "None"),  SERVICECATEGORY=columnifexists("SERVICECATEGORY", "None"),  SERVICEID=columnifexists("SERVICEID", "None"),  SERVICEMSGID=columnifexists("SERVICEMSGID", "None"),  
    SERVICENAME=columnifexists("SERVICENAME", "None"),  SERVICETYPE=columnifexists("SERVICETYPE", "None"),  SERVICEUSER=columnifexists("SERVICEUSER", "None"),  SERVICEUSERTYP=columnifexists("SERVICEUSERTYP", "None"),  CLIENTNAME=columnifexists("CLIENTNAME", "None"),  
    LOGINREASON=columnifexists("LOGINREASON", "None"),  OSUSR=columnifexists("OSUSR", "None"),  ROLE=columnifexists("ROLE", "None"),  USERJOB=columnifexists("USERJOB", "None"),  E3MID=columnifexists("E3MID", "None"),  MASKMODE=columnifexists("MASKMODE", "None"),  
    APP=columnifexists("APP", "None"),  CTXT=columnifexists("CTXT", "None"),  E5FID=columnifexists("E5FID", "None"),  FILENAME=columnifexists("FILENAME", "None"),  ACTNDTL=columnifexists("ACTNDTL", "None"),  AUDACTN=columnifexists("AUDACTN", "None"),  
    DLGLINE=columnifexists("DLGLINE", "None"),  PULID=columnifexists("PULID", "None"),  TIMEOUT=columnifexists("TIMEOUT", "None"),  NEWUSER=columnifexists("NEWUSER", "None"),  PREVUSER=columnifexists("PREVUSER", "None"),  MYACCT=columnifexists("MYACCT", "None"),  
    MYCACCT=columnifexists("MYCACCT", "None"),  EAR=columnifexists("EAR", "None"),  SSUPATLNAME=columnifexists("SSUPATLNAME", "None"),  MSGID=columnifexists("MSGID", "None"),  SIGNUPMETHOD=columnifexists("SIGNUPMETHOD", "None"),  PWDATTEMPTCNT=columnifexists("PWDATTEMPTCNT", "None"),  
    AllIPs=columnifexists("AllIPs", "None"),  ecIP=columnifexists("ecIP", "None"),  wsIP=columnifexists("wsIP", "None")
```